<launch>
  <include file="$(find autorally_core)/launch/hardware.machine" />

  <group ns="safeSpeed">
    <param name="maxSafeSpeed" value="25" />
    <param name="Timeout" value="5.0" />
    <param name="KP" value="5.0" />
    <param name="KD" value="0.5" />
    <param name="KI" value="0.2" />
    <param name="IResetRate" value="0.1" />
    <param name="ThrottleOffset" value="17" />
    <param name="Multiplier" value="0.012" /> <!--Assuming max speed of 40mph (17.88m/s) -->
  </group>

  <node pkg="nodelet" type="nodelet" name="ServoInterface" args="load autorally_core/ServoInterface autorally_core_manager" output="screen">
<!--  <node name="servoInterface" pkg="autorally_core" type="servoInterface" machine="muri-laptop" output="screen"> -->
    <param name="controllerType" value="micro"/>
    <param name="servoCommandRate" value="75" />
    <param name="throttleBrakeCoupled" value="true" />
    <param name="servoCommandMaxAge" value="0.2" />

    <rosparam param="servos" command="load" file="$(env AR_CONFIG_PATH)/arChassisConfig_$(env AR_CHASSIS).yaml" />
    <!-- <rosparam param="throttleCalibration" command="load" file="$(env AR_CONFIG_PATH)/throttlePositionCalibration.yaml" /> -->

    <rosparam param="servoCommandProirities" command="load" file="$(env AR_CONFIG_PATH)/servoCommandPriorities.yaml" />
    <!--configure settings for 57600 baud, 8N1 -->
    <param name="port" value="/dev/arServoController" />
    <param name="serialBaud" value="57600" />
    <param name="serialDataBits" value="8" />
    <param name="serialParity" value="none" />
    <param name="serialStopBits" value="1" />
    <param name="serialHardwareFlow" value="false" />
    <param name="serialSoftwareFlow" value="false" />
  </node>

</launch>
